chef/cookbooks/cpe_munki/resources/cpe_munki_install.rb

```diff
@@ -15,108 +15,101 @@
-    cookbook_file item do
-      not_if { ::File.exist?('/Library/CPE/tags/munki_test') }
-      action :create
-      group 'wheel'
-      mode '755'
-      owner 'root'
-      source "munki/admin/#{munki_admin_version}/#{item}"
-    end
+  def write_munki_lastcheckdate(time)
+    execute "defaults write #{managed_installs} LastCheckDate '#{time}'"
   end
 
-  munki_ld_version = munki['munki_launchd_version']
+  def remediate?
+    return unless node['cpe_munki']['auto_remediate']
+    # There is a chance that this is prior to the first run,
+    # file might not be present.
+    return unless ::File.exist?("#{managed_installs}.plist")
+    # getting value for the last munki check-in
+    last_check_date = read_munki_lastcheckdate
+    # if it's greater than the remediate_window since munki last ran.
+    today = Date.today
+    remediate_window = node['cpe_munki']['auto_remediate'].to_i
+    return unless (today - last_check_date).to_i > remediate_window
 
-  munki['munki_launchd_folders'].each do |item|
-    directory item do
-      action :create
-      group 'wheel'
-      mode 0o755
-      owner 'root'
+    # and the pkg is installed
+    node['cpe_munki']['munki_version_to_install'].to_h.each do |pkg, _opts|
+      receipt = "com.googlecode.munki.#{pkg}"
+      next if Mixlib::ShellOut.new(
+        "/usr/sbin/pkgutil --pkg-info='#{receipt}'",
+      ).run_command.error?
+      # forget the pkg
+      execute "/usr/sbin/pkgutil --forget #{receipt}"
     end
+    write_munki_lastcheckdate(Time.now)
   end
+end
 
-  munki['munki_launcha_files'].each do |item|
-    launcha_path = "/Library/LaunchAgents/#{item}"
-    cookbook_file launcha_path do
-      not_if { ::File.exist?('/Library/CPE/tags/munki_test') }
-      action :create
-      group 'wheel'
-      mode '755'
-      owner 'root'
-      source "munki/launchd/#{munki_ld_version}/Library/LaunchAgents/#{item}"
+action :install do
+  return unless node['cpe_munki']['install']
+  m = 'com.googlecode.munki'
+  remediate?
+  node['cpe_munki']['munki_version_to_install'].to_h.each do |pkg, opts|
+    package_version = opts['version']
+    package_name = "munkitools_#{pkg}-#{package_version}"
+    cpe_remote_pkg "munkitools_#{pkg}" do # ~FC037
+      app 'munkitools'
+      pkg_name package_name
+      checksum opts['checksum']
+      receipt "com.googlecode.munki.#{pkg}"
+      version package_version
+      if pkg.include?('launchd')
+        notifies :restart, "launchd[#{m}.logouthelper]"
+        notifies :restart, "launchd[#{m}.managedsoftwareupdate-check]"
+        notifies :restart, "launchd[#{m}.managedsoftwareupdate-manualcheck]"
+        notifies :restart, "launchd[#{m}.authrestartd]"
+        notifies :restart, "launchd[#{m}.managedsoftwareupdate-install]"
+        notifies :restart, "launchd[#{m}.app_usage_monitor]"
+        notifies :restart, "launchd[#{m}.munki-notifier]"
+        notifies :restart, "launchd[#{m}.ManagedSoftwareCenter]"
+      end
```

Comment:
This series of `notifies` will error on initial chef runs if munki isn't deployed. Doing a second chef run will clean up the errors and allow the chef run to continue. Not sure the best solution to guard against this.

[error.txt](https://github.com/facebook/IT-CPE/files/1229216/error.txt)